fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -1584,16 +1573,49 @@ Future<std::vector<std::pair<TenantName, TenantMapEntry>>> listTenants(
     TenantName begin,
     TenantName end,
     int limit,
+    int offset = 0,
     std::vector<TenantState> filters = std::vector<TenantState>()) {
 	state Reference<typename DB::TransactionT> tr = db->createTransaction();
 
 	loop {
 		try {
 			tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
 			tr->setOption(FDBTransactionOptions::READ_LOCK_AWARE);
-			std::vector<std::pair<TenantName, TenantMapEntry>> tenants =
-			    wait(listTenantsTransaction(tr, begin, end, limit, filters));
-			return tenants;
+			if (offset == 0 && filters.empty()) {
+				std::vector<std::pair<TenantName, TenantMapEntry>> tenants =
+				    wait(listTenantsTransaction(tr, begin, end, limit));
+				return tenants;
+			}
+			tr->setOption(FDBTransactionOptions::RAW_ACCESS);
+
+			state KeyBackedRangeResult<std::pair<TenantName, TenantMapEntry>> results =
+			    wait(ManagementClusterMetadata::tenantMetadata().tenantMap.getRange(tr, begin, end, limit));
```

Comment:
The limit here could probably be at least `limit + offset`, but we may want to have some minimum value on this if we specify filters. Otherwise, a limit of something like 1 is going to move very slowly through the results.